google_bigquery/models/ir_model_fields_bigquery.py

```diff
@@ -0,0 +1,175 @@
+from google.cloud import bigquery
+
+from odoo import api, fields, models
+from odoo.tools.float_utils import float_repr
+
+
+class IrModelFieldsBigQuery(models.Model):
+    _name = "ir.model.fields.bigquery"
+    _description = "Field BigQuery Settings"
+
+    enabled = fields.Boolean(default=True)
+    nested = fields.Boolean(
+        default=False, help="This will create a struct field in Google BigQuery"
+    )
+    field_id = fields.Many2one(
+        comodel_name="ir.model.fields", ondelete="cascade", required=True
+    )
+    field_relation = fields.Char(related="parent_id.field_id.relation")
+    field_type = fields.Selection(related="field_id.ttype")
+    model_bigquery_id = fields.Many2one(
+        comodel_name="ir.model.bigquery", string="Model"
+    )
+    bigquery_type = fields.Selection(
+        selection=[
+            ("BOOL", "BOOL"),
+            ("BYTES", "BYTES"),
+            ("DATE", "DATE"),
+            ("DATETIME", "DATETIME"),
+            ("INTEGER", "INTEGER"),
+            ("NUMERIC", "NUMERIC"),
+            ("STRING", "STRING"),
+            ("STRUCT", "STRUCT"),
+        ],
+        compute="_compute_bigquery_field_info",
+    )
+    bigquery_mode = fields.Selection(
+        selection=[
+            ("NULLABLE", "NULLABLE"),
+            ("REQUIRED", "REQUIRED"),
+            ("REPEATED", "REPEATED"),
+        ],
+        compute="_compute_bigquery_field_info",
+    )
+    parent_id = fields.Many2one(
+        comodel_name="ir.model.fields.bigquery",
+        ondelete="cascade",
+        help="In case of a struct field",
+    )
+    child_ids = fields.One2many(
+        comodel_name="ir.model.fields.bigquery", inverse_name="parent_id"
+    )
```

Comment:
If you have a parent-child structure, you should also have `parent_path` for fast searching?